lib/SIL/IR/SILInstruction.cpp

```diff
@@ -1519,6 +1519,27 @@ MultipleValueInstruction *MultipleValueInstructionResult::getParent() {
   return reinterpret_cast<MultipleValueInstruction *>(value);
 }
 
+/// Returns true if evaluation of this node may cause suspension of an
+/// async task.
+bool SILNode::maySuspend() const {
+  // await_async_continuation always suspends the current task.
+  if (isa<AwaitAsyncContinuationInst>(this))
+    return true;
+  
+  // Fully applying an async function may suspend the caller.
+  if (auto i = dyn_cast<SILInstruction>(this)) {
+    if (isa<FullApplySite>(i)) {
```

Comment:
Just a minor suggestion, but I think you can do 
```suggestion
    if (auto FAI = FullApplySite::isa(&i)) {
```

and use `FAI.getOrigCalleeType()` which should simplify some code below.

Reply:
Thanks. The `dyn_cast` half-implementation seems like a bit of an attractive nuisance.